ingestion/src/metadata/distributed/argo_orchestrator.py

```diff
@@ -0,0 +1,313 @@
+    """
+
+    def __init__(
+        self,
+        namespace: str = "default",
+        service_account: Optional[str] = None,
+        image: str = "openmetadata/ingestion:latest",
+    ):
+        self.namespace = namespace
+        self.service_account = service_account
+        self.image = image
+
+    def generate_workflow(
+        self,
+        workflow_config: Dict[str, Any],
+        source: DiscoverableSource,
+        parallelism: int = 50,
+        retry_limit: int = 3,
+    ) -> Dict[str, Any]:
+        """
+        Generate complete Argo Workflow definition.
+
+        Args:
+            workflow_config: OpenMetadata workflow configuration
+            source: Source implementing DiscoverableSource
+            parallelism: Maximum parallel worker pods
+            retry_limit: Number of retries for failed entities
+
+        Returns:
+            Argo Workflow definition as dict
+        """
+        entity_types = source.get_parallelizable_entity_types()
+        dependencies = source.get_entity_type_dependencies()
+
+        config_b64 = self._encode_config(workflow_config)
+
+        workflow = {
+            "apiVersion": "argoproj.io/v1alpha1",
+            "kind": "Workflow",
+            "metadata": {
+                "generateName": f"{workflow_config.get('source', {}).get('serviceName', 'om')}-ingestion-",
+                "namespace": self.namespace,
+            },
+            "spec": {
+                "entrypoint": "main",
+                "arguments": {
+                    "parameters": [
+                        {"name": "workflow-config", "value": config_b64},
+                        {"name": "parallelism", "value": str(parallelism)},
+                    ]
+                },
+                "templates": [
+                    self._build_main_dag(entity_types, dependencies, parallelism),
+                    self._build_discovery_template(),
+                    self._build_worker_template(retry_limit),
+                ],
+            },
+        }
+
+        if self.service_account:
+            workflow["spec"]["serviceAccountName"] = self.service_account
+
+        return workflow
+
+    def _build_main_dag(
+        self,
+        entity_types: List[str],
+        dependencies: Dict[str, List[str]],
+        parallelism: int,
+    ) -> Dict[str, Any]:
+        """Build main DAG template with discovery and processing phases"""
+        tasks = []
+
+        # Discovery tasks - one per entity type
+        for entity_type in entity_types:
+            # Replace underscores with hyphens for Argo task names
+            task_name = entity_type.replace("_", "-")
+            tasks.append(
+                {
+                    "name": f"discover-{task_name}",
+                    "template": "discover",
+                    "arguments": {
+                        "parameters": [{"name": "entity-type", "value": entity_type}]
+                    },
+                }
+            )
+
```

Comment:
What do you mean here 1 discovery task per entity type? Would a workflow have only 1 entity type?